### ISSUE TITLE
pipeline for fetch.map_geoms

### ISSUE THREAD

**Issue by jordansread** [USGS-VIZLAB/vizstorm#23]
Spinoff of #10, but only for the fetch step.

**Comment by dblodgett-usgs** [comment]
Starting from: 
https://github.com/USGS-VIZLAB/hurricane-maria/blob/master/scripts/fetch/fetch.islands.R
https://github.com/USGS-VIZLAB/hurricane-irma/blob/master/scripts/fetch/fetch.states.R
https://github.com/USGS-VIZLAB/hurricane-maria/blob/master/scripts/fetch/utils.R
https://github.com/dblodgett-usgs/uswb-viz/blob/master/scripts/process/process_state_map_data.R#L60

These currently focus on getting data from the maps package. I'll start there and we can generalize unless there's an argument to be made to use anything else. I'd lean toward working with what we know to get an end to end start then introducing more change once things are serially complete.

This work will support culling whole features that are out of the view box.

**Comment by jordansread** [comment]
To cull the features, are you going to use `viewbox_limits` as a post process to things like `states` or pre-process similar to fetch.islands use of `xlim = c(275, 300), ylim = c(16, 30),...`?

**Comment by dblodgett-usgs** [comment]
Currently, I have both... 
From the `@examples`
```
#' islands <- get_map_data_sf(crs = "+init=epsg:3857", database = 'mapdata::world2Hires', 
#'                            regions = "(?!USA)", xlim = c(275, 300), ylim = c(16, 30))
#'
#'# A polygon in a particular projection can also be used.
#' bbox <- bbox_to_polygon(c(-87, 21, -70, 34), return_crs = "+init=epsg:3857")
#' storm_counties <- get_map_data_sf(database = "counties", crs = "+init=epsg:3857", within = bbox)
```
p.s. sf:: is sweet! `map_data_subset <- map_data[sf::st_intersects(map_data, within, sparse = F),]` does the subset thing like:
<img width="260" alt="screen shot 2017-10-28 at 2 08 46 pm" src="https://user-images.githubusercontent.com/1492803/32137653-8cc970a0-bbe9-11e7-8de5-21c49b7dac5c.png">
and `map_data_clip <- sf::st_intersection(map_data, within)
` does the clip thing.
<img width="275" alt="screen shot 2017-10-28 at 2 09 30 pm" src="https://user-images.githubusercontent.com/1492803/32137658-ad989c20-bbe9-11e7-9559-115b07f06705.png">

Any reason to leave one or the other out subset methods? Note I do NOT think we should "clip" in this stage.

**Comment by jordansread** [comment]
what does your viz.yaml look like to specify those arguments? 

can you clarify what you mean by `Note I do NOT think we should "clip" in this stage.` does that mean you want to return all US counties if we are fetching counties?

**Comment by dblodgett-usgs** [comment]
Working on the viz.yml now. Just a utility function so far. 

I think we should do a spatial join, keeping whole features that intersect (what I'm calling subset above). I think we should not apply a clip to the features as in what I'm calling "clip" above. I think that's in line with your comments over in #10?

**Comment by jordansread** [comment]
Agreed. so you are planning on filtering the result to just what even partially overlaps with the viewbox, and saving the actual modification of the geoms (`clipping`) for later (process)? makes sense

**Comment by dblodgett-usgs** [comment]
yeppers.

**Comment by dblodgett-usgs** [comment]
My proposal for the viz.yml is to have each type of feature be it's own super-simple-fetcher as in Irma. https://github.com/USGS-VIZLAB/hurricane-irma/blob/master/scripts/fetch/fetch.states.R

So there'll be a menu of fetchers that you'll be able to include in your viz.yml or not depending on the needs of the viz. Make sense?

**Comment by jordansread** [comment]
fixed in #26
